File: AmsterdamEats.Maui/AmsterdamEats.Maui/Models/ZenchefAvailability.cs
using System.Text.Json.Serialization;

namespace AmsterdamEats.Models;

public class ZenchefShift
{
    [JsonPropertyName("id")] public int Id { get; set; }
    [JsonPropertyName("name")] public string Name { get; set; } = string.Empty;
    [JsonPropertyName("closed")] public bool Closed { get; set; }
    [JsonPropertyName("possible_guests")] public List<int> PossibleGuests { get; set; } = [];
    [JsonPropertyName("schedule")] public ZenchefSchedule Schedule { get; set; } = new();
}

public class ZenchefSchedule
{
    [JsonPropertyName("date")] public string Date { get; set; } = string.Empty;
}

public class ZenchefSlot
{
    public int Id { get; set; }
    public string ShiftName { get; set; } = string.Empty;
    public string Date { get; set; } = string.Empty;
}

File: AmsterdamEats/AmsterdamEats/Models/ZenchefAvailability.swift
import Foundation

struct ZenchefShift: Decodable {
    let id: Int
    let name: String
    let closed: Bool
    let possibleGuests: [Int]
    let schedule: Schedule

    struct Schedule: Decodable {
        let date: String
    }

    private enum CodingKeys: String, CodingKey {
        case id, name, closed, schedule
        case possibleGuests = "possible_guests"
    }
}

struct ZenchefSlot: Identifiable {
    let id: Int
    let shiftName: String
    let date: String
}
